dns_check/datadog_checks/dns_check/dns_check.py

```diff
@@ -62,29 +62,29 @@ def _get_resolver(self):
         return resolver
 
     def check(self, _):
-        resolver = self._get_resolver()
-
         # Perform the DNS query, and report its duration as a gauge
-        t0 = get_precise_time()
+        resolver = self._get_resolver()
 
         try:
             self.log.debug('Querying "%s" record for hostname "%s"...', self.record_type, self.hostname)
             if self.record_type == "NXDOMAIN":
                 try:
+                    t0 = get_precise_time()
                     resolver.query(self.hostname)
                 except dns.resolver.NXDOMAIN:
-                    pass
+                    response_time = get_precise_time() - t0
```

Comment:
Why do we still want to get the response time from a record we fail to query?

Reply:
Discussed this on the side. But this was to keep the original logic in place. The idea was that the `NXDOMAIN` exception is expected and the actual thing we're timing here. Essentially how long it takes to get the `NXDOMAIN` exception.